app/models/manageiq/providers/amazon/cloud_manager/ssa_agent_manager.rb

```diff
@@ -0,0 +1,326 @@
+
+      # create security group if not exist
+      security_group = ec2.create_security_group({
+        group_name: group_name,
+        description: 'Security group for MIQ SSA Agent',
+        vpc_id: ec2.client.describe_vpcs.vpcs[0].vpc_id
+      })
+
+      security_group.authorize_ingress({
+        ip_permissions: [{
+          ip_protocol: 'tcp',
+          from_port: 22,
+          to_port: 22,
+          ip_ranges: [{
+            cidr_ip: '0.0.0.0/0'
+          }]}]
+      })
+
+      security_group.authorize_ingress({
+        ip_permissions: [{
+          ip_protocol: 'tcp',
+          from_port: 80,
+          to_port: 80,
+          ip_ranges: [{
+            cidr_ip: '0.0.0.0/0'
+          }]}]
+      })
+
+      security_group.authorize_ingress({
+        ip_permissions: [{
+          ip_protocol: 'tcp',
+          from_port: 443,
+          to_port: 443,
+          ip_ranges: [{
+            cidr_ip: '0.0.0.0/0'
+          }]}]
+      })
+
+      security_group.group_id
+    end
+  end
+
+  # Get Key Pair for SSH. Create a new one if not exists. 
+  def get_key_pair(pair_name = MIQ_SSA)
+    kp = Authentication.where(name: pair_name)
+    return kp[0] if kp.length > 0
+
+    ManageIQ::Providers::CloudManager::AuthKeyPair.create_key_pair(ext_management_system.id,
+      { :key_name => pair_name })
+  end
+
+  def create_pem_file(pair_name = MIQ_SSA)
+    kp = get_key_pair(pair_name)
+    pem_file_name = pair_name+".pem"
+    File.open(pem_file_name, 'w') {|f| f.write(kp.auth_key) }
+    File.chmod(0400, pem_file_name)
+    pem_file_name
+  end
+
+  def agent_startup_script(script_name = '/tools/amazon_ssa_extract.rb')
+    ssa_root = Gem.loaded_specs['amazon_ssa_support'].full_gem_path
+    ssa_script = ssa_root + script_name
+    ssa_cmd = "ruby #{ssa_script}"
+  end
+
+  def default_settings
+    "echo -e '---\n:log_level: #{log_level}\n:region: #{region}\n:request_queue: #{request_queue}\n:reply_queue: #{reply_queue}\n:ssa_bucket: #{ssa_bucket}\n:reply_prefix: #{reply_prefix}\n:log_prefix: #{log_prefix}\n:heartbeat_prefix: #{heartbeat_prefix}\n:heartbeat_interval: i#{heartbeat_interval}\n' > default_ssa_config.yml"
+  end
+
+  def create_user_data
+    script = '#!/bin/bash
+yum -y update > /var/log/miq_ssa_deploy.log 2>&1
+yum -y install git-core zlib zlib-devel gcc-c++ patch readline readline-devel libyaml-devel libffi-devel openssl-devel make bzip2 autoconf automake libtool bison curl sqlite-devel postgresql-devel >> /var/log/miq_ssa_deploy.log 2>&1
+git clone https://github.com/rbenv/rbenv.git ~/.rbenv >> /var/log/miq_ssa_deploy.log 2>&1
+export HOME="/root" >> /var/log/miq_ssa_deploy.log 2>&1
+echo \'export PATH="$HOME/.rbenv/bin:$PATH"\' >> ~/.bash_profile
+echo \'eval "$(/root/.rbenv/bin/rbenv init -)"\' >> ~/.bash_profile
+source ~/.bash_profile >> /var/log/miq_ssa_deploy.log 2>&1
+git clone git://github.com/sstephenson/ruby-build.git ~/.rbenv/plugins/ruby-build >> /var/log/miq_ssa_deploy.log 2>&1
```

Comment:
Is this path right?  Why are we using "sstephenson's" ruby-build instead of the one in rbenv like 5 lines above?